contracts/FungibleTokenMetadataViews.cdc

```diff
@@ -0,0 +1,168 @@
+            description: String,
+            externalURL: MetadataViews.ExternalURL,
+            logos: MetadataViews.Medias,
+            socials: {String: MetadataViews.ExternalURL}
+        ) {
+            self.name = name
+            self.symbol = symbol
+            self.description = description
+            self.externalURL = externalURL
+            self.logos = logos
+            self.socials = socials
+        }
+    }
+
+    /// Helper to get FTDisplay in a way that will return a typed optional
+    /// 
+    /// @param viewResolver: A reference to the resolver resource
+    /// @return A optional FTDisplay struct
+    ///
+    pub fun getFTDisplay(_ viewResolver: &{MetadataViews.Resolver}): FTDisplay? {
+        if let view = viewResolver.resolveView(Type<FTDisplay>()) {
+            if let v = view as? FTDisplay {
+                return v
+            }
+        }
+        return nil
+    }
+
+    /// View to expose the information needed store and interact with a FT vault.
+    /// This can be used by applications to setup a FT vault with proper 
+    /// storage and public capabilities.
+    ///
+    pub struct FTVaultData {
+        /// Path in storage where this FT vault is recommended to be stored.
+        pub let storagePath: StoragePath
+
+        /// Public path which must be linked to expose public capabilities of
+        /// this FT, including standard FT interfaces and metadataviews 
+        /// interfaces
+        pub let publicPath: PublicPath
+
+        /// Private path which should be linked to expose the provider
+        /// capability to withdraw funds from the vault
+        pub let providerPath: PrivatePath
+
+        /// Type that should be linked at the aforementioned public path. This 
+        /// is normally a restricted type with many interfaces. Notably the 
+        /// `FungibleToken.Balance`, `FungibleToken.Receiver`, and  
+        /// `MetadataViews.Resolver` interfaces are required.
+        pub let publicLinkedType: Type
+
+        /// Type that should be linked at the aforementioned private path. This 
+        /// is normally a restricted type with at a minimum the  
+        /// `FungibleToken.Provider` interface
+        pub let providerLinkedType: Type
+
+        /// Function that allows creation of an empty FT vault that is intended
+        /// to store the funds.
+        pub let createEmptyVault: ((): @FungibleToken.Vault)
+
+        init(
+            storagePath: StoragePath,
+            publicPath: PublicPath,
+            providerPath: PrivatePath,
+            publicLinkedType: Type,
+            providerLinkedType: Type,
+            createEmptyVaultFunction: ((): @FungibleToken.Vault)
+        ) {
+            pre {
+                publicLinkedType.isSubtype(of: Type<&{FungibleToken.Receiver, FungibleToken.Balance, MetadataViews.Resolver}>()): "Public type must include FungibleToken.Receiver, FungibleToken.Balance and MetadataViews.Resolver interfaces."
```

Comment:
I don't think this is going to work with the way we've designed the paths above. The balance and receiver interfaces are linked at different public paths, and currently, the resolver is also at a different path. I think that the resolver should be linked at the same path as balance and then we'll probably need to update this view to have two public paths (the metadata path and the receiver path). Make sure there is a test case for this too

Reply:
Actually now there's only one test case using this transaction https://github.com/onflow/flow-ft/blob/alilloig/example-token-metadata-views/transactions/setup_account_from_vault_reference.cdc

I just realise I should extend the case a little bit more and try to actually use the freshly new created `Vault`, but basically `<&{FungibleToken.Receiver, FungibleToken.Balance, MetadataViews.Resolver}>` are exposed at `/public/exampleTokenReceiver`.

I should improve consistency around how the paths are used probably.

Reply:
yeah, I would maybe expose all the interfaces at the receiver path and the balance path just to be thorough

Reply:
I refurbished everything about the public paths on the `FTVaultData` view, you can see it here https://github.com/onflow/flow-ft/pull/89/commits/089b9fbd31d63bd09ed7d10cfd5af5dc9f0e5364 and it has already been updated for using it that way on the ExampleToken

Reply:
So I didn't expose all interfaces on both paths (that could be done easily) but create two separate fields on the `FTVaultData` one for the receiver path and other for the metadata path, with its correspondent linked types